packages/playwright-core/src/common/socksProxy.ts

```diff
@@ -375,7 +375,7 @@ export class SocksProxyHandler extends EventEmitter {
 
   async socketRequested({ uid, host, port }: SocksSocketRequestedPayload): Promise<void> {
     if (host === 'local.playwright')
-      host = 'localhost';
```

Comment:
This may backfire if the host does custom mapping for `localhost` name, maybe we can update our test server instead?